src/coreclr/nativeaot/BuildIntegration/Microsoft.DotNet.ILCompiler.props

```diff
@@ -10,10 +10,11 @@ Copyright (c) .NET Foundation. All rights reserved.
 ***********************************************************************************************
 -->
 <Project>
-
   <PropertyGroup>
+    <!-- Set the publishAot property to true if not set-->
+    <PublishAot Condition="'$(PublishAot)' == ''">true</PublishAot>
```

Comment:
If I understand it correctly, this was needed because we do the ".targets included from a .props file" antipattern - I think we discussed this in one of the past pull request - to do the real fix, we likely need to break up Microsoft.DotNet.ILCompiler.targets into Microsoft.DotNet.ILCompiler.targets and Microsoft.DotNet.ILCompiler.props as needed.

(The convention is that .props are included before the user project, .targets get included after the user provided stuff - we would see PublishAot being set there if the user wants it.)

Reply:
Most of the stuff that is already in .targets should be in .targets. We wouldn't probably have much (if anything?) in the props.

Reply:
Ah, you're right. I saw .targets in a .props and I thought it's an `Import`, but it's just storing it in a property for later.